### PR TITLE
OSASINFRA-3612: OpenStack: Use csi-operator jobs to build and test manila CSI

### PR DESCRIPTION
**Pull request by MaysaMacedo**
As part of the process to migrate the manila CSI operator into the csi-operator, this PR moves the images built to be based on the legacy folder under csi-operator.

here is the PR for shipping the image from the new operator https://github.com/openshift-eng/ocp-build-data/pull/5403

depends on https://github.com/openshift/csi-operator/pull/280

### CODE REVIEW

**Comment by MaysaMacedo** [comment]
/hold

until dependent PR is merged

**Review by mandre — COMMENTED** [review]
This needs a rebase after https://github.com/openshift/release/pull/56812 has merged.

**Review by mandre — COMMENTED** [review]
/lgtm
/approve

**Comment by mandre** [comment]
/pj-rehearse pull-ci-openshift-csi-operator-master-e2e-openstack-manila-csi

**Comment by MaysaMacedo** [comment]
/pj-rehearse pull-ci-openshift-csi-driver-manila-operator-release-4.17-e2e-openstack-csi

File: ci-operator/config/openshift/csi-operator/openshift-csi-operator-release-4.19.yaml [hunk 1]
```diff
@@ -308,7 +320,7 @@ tests:
     workflow: openshift-e2e-azure-csi-extended
 - as: e2e-openstack
   optional: true
-  run_if_changed: ^(Dockerfile\.openstack-cinder|legacy/openstack-cinder-csi-driver-operator/.*)
+  run_if_changed: ^(Dockerfile\.openstack-cinder|legacy/openstack-cinder-csi-driver-operator/.*|Dockerfile\.openstack-manila|legacy/csi-driver-manila-operator/.*)
   steps:
```
Comment (gnufied) [inline ci-operator/config/openshift/csi-operator/openshift-csi-operator-release-4.19.yaml:324]: I realized that path separator needs to be escaped. We should escaped them before as well.

Comment (MaysaMacedo) [reply ci-operator/config/openshift/csi-operator/openshift-csi-operator-release-4.19.yaml:324]: Just pushed. Is that what you had in mind?

Comment (gnufied) [reply ci-operator/config/openshift/csi-operator/openshift-csi-operator-release-4.19.yaml:324]: No. Sorry - I meant - `^(Dockerfile\.openstack-cinder|legacy\/openstack-cinder-csi-driver-operator\/.*|Dockerfile\.openstack-manila|legacy\/csi-driver-manila-operator\/.*)`

**Comment by gnufied** [comment]
/pj-rehearse pull-ci-openshift-csi-operator-master-e2e-openstack-manila-csi

**Review by mandre — COMMENTED** [review]
/lgtm

**Pull request merged**